lib/util/urltils.js

```diff
@@ -162,6 +162,31 @@ module.exports = {
     }
   },
 
+  /**
+   * Obfuscates path parameters with regex from config
+   *
+   * @param {Config} config The configuration containing the regex
+   * @param {string} path The path to be obfuscated
+   * @returns {string} The obfuscated path or the original path
+   */
+  obfuscatePath: function obfuscatePath(config, path) {
+    // TODO: handle SyntaxError: Invalid regular expression
+    if (typeof path !== 'string') {
+      return path
+    }
+
+    const { enabled, regex } = config.url_obfuscation
+    if (!enabled || !regex) {
+      return path
+    }
+
+    const { pattern, flags, replacement } = regex
+    const f = !flags ? '' : flags
+    const r = !replacement ? '' : replacement
+    const regexPattern = new RegExp(pattern, f)
+    return path.replace(regexPattern, `/${r}`)
```

Comment:
```suggestion
    const { pattern, flags = '', replacement = '' } = regex
    const regexPattern = new RegExp(pattern, flags)
    return path.replace(regexPattern, `/${replacement}`)
```

Reply:
I was thinking about the same, this will require me to change default values from `null` to an empty string `''` in the `/lib/config/default.js` file. I think it's a better solution overall. I went with your initial suggestion to use `null` values in defaults and from what I'm seeing we can simply use `''` instead.

Reply:
ah yea, we should set these to empty strings in `lib/config/default.js`, my bad